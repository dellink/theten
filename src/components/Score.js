import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default function Score({ currentValue, record }) {
  return (
    <View style={styles.container}>
      <View style={styles.line}>
        <Text style={styles.text}>Score:</Text>
        <Text style={styles.value}>{currentValue}</Text>
      </View>
      <View style={styles.line}>
        <Text style={styles.text}>Record:</Text>
        <Text style={styles.value}>{record}</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#03A9F4',
    borderColor: '#0288D1',
    borderRadius: 4,
    borderWidth: 1,
    flexDirection: 'column',
    minWidth: 120,
    padding: 10,
    paddingBottom: 0,
  },
  line: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 10,
  },
  text: {
    color: '#B3E5FC',
    fontFamily: 'Verdana',
    fontSize: 14,
    fontWeight: 'bold',
    marginRight: 10,
    textAlign: 'center',
  },
  value: {
    color: 'white',
    fontFamily: 'Verdana',
    fontSize: 14,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
