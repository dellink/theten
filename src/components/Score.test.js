import React from 'react';
import { render } from '@testing-library/react-native';
import Score from './Score';

describe(Score.name, () => {
  it('should render correctly', () => {
    expect(render(<Score record={0} currentValue={0} />).asJSON()).toMatchSnapshot();
  });
});
