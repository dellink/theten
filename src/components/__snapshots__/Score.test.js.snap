// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score should render correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#03A9F4",
        "borderColor": "#0288D1",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "column",
        "minWidth": 120,
        "padding": 10,
        "paddingBottom": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#B3E5FC",
            "fontFamily": "Verdana",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginRight": 10,
            "textAlign": "center",
          }
        }
      >
        Score:
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Verdana",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        0
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#B3E5FC",
            "fontFamily": "Verdana",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginRight": 10,
            "textAlign": "center",
          }
        }
      >
        Record:
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Verdana",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        0
      </Text>
    </View>
  </View>
</View>
`;
